### PR TITLE
Improved tool parameter error reporting

### PR DESCRIPTION
**Pull request by bernt-matthias**
To simplify tool testing its of advantage to know which parameter caused a problem.

### CODE REVIEW

**Comment by mvdbeek** [comment]
Nice!

File: lib/galaxy/tools/parameters/basic.py [hunk 1]
```diff
@@ -1862,10 +1862,10 @@ def do_validate(v):
 
         if self.min is not None:
             if self.min > dataset_count:
-                raise ValueError("At least %d datasets are required." % self.min)
+                raise ValueError("At least %d datasets are required for %s" % (self.min,self.name))
```
Comment (mvdbeek) [inline lib/galaxy/tools/parameters/basic.py]:
```suggestion
                raise ValueError("At least %d datasets are required for %s" % (self.min, self.name))
```

**Comment by martenson** [comment]
thank you @bernt-matthias !

**Comment by martenson** [comment]
xref backport: https://github.com/galaxyproject/galaxy/pull/8203

**Pull request merged**